src/nccl_ofi_net.c

```diff
@@ -1543,8 +1579,25 @@ static ncclResult_t ofi_accept(void *listenComm, void **recvComm)
 			goto exit;
 	}
 
+	/* non NULL memory buffer means connection was local */
+	if (conn_info.memory_buffer) {
+		NCCL_OFI_TRACE(NCCL_NET, "Accept from self");
+		/* process postponed completions */
+		do {
+			ret = nccl_ofi_progress(nccl_ofi_component[conn_info.req->dev]);
+			if (OFI_UNLIKELY(ret != 0))
+				goto error;
+		} while (conn_info.req->state != NCCL_OFI_REQ_COMPLETED);
+		free_nccl_ofi_req(conn_info.req, false);
+		free(conn_info.memory_buffer);
+		NCCL_OFI_TRACE(NCCL_NET, "Completions processed");
```

Comment:
`free(conn_info)`?

Reply:
> `free(conn_info)`?

Nono, conn_info was not allocated from heap, at receiver side it is just a buffer on stack, so no need to free it.

Reply:
Key point with conn_info buffers is that they are not the same, they are two separate buffers, one of them (send side) needs to live longer than buffer at receiver side, but at receiver side we deal with different buffer thus cannot just check its address.